Formula/nbdime.rb

```diff
@@ -17,9 +17,11 @@ class Nbdime < Formula
 
   depends_on "python@3.9"
 
-  resource "appnope" do
-    url "https://files.pythonhosted.org/packages/26/34/0f3a5efac31f27fabce64645f8c609de9d925fe2915304d1a40f544cff0e/appnope-0.1.0.tar.gz"
-    sha256 "8b995ffe925347a2138d7ac0fe77155e4311a0ea6d6da4f5128fe4b3cbe5ed71"
+  on_macos do
+    resource "appnope" do
```

Comment:
Does does our current `brew bump-formula-pr`/`brew update-python-resources` work well with `on_macos` logic?

Reply:
I don't know. I never used `brew update-python-resources`. But we have some other formulae that will have the same issue, mainly perl-based ones, which need to ship a lot more linux resources compared to the macOS ones.

Reply:
In the case of Python, when a user does `brew bump-formula-pr` on a PyPi URL, it rewrites the resource blocks automatically.

I don't think we support this yet for any other language like Perl, so we don't need to worry about that yet.

---

I just tried adding `on_macos` to `appnope` and running `brew update-python-resources` and it failed with:
```
Error: Unable to update resource blocks for "nbdime" automatically. Please update them manually.
```
This probably means the update here would require a multi-step approach like:
1. comment out macos_only block
2. `brew update-python-resources`
3. re-add macos_only block and check for updates there.

---

I guess the ideal solution would be to improve the bump logic to handle this so that `brew bump-formula-pr` can be run as single command.

The currently working alternative is to split up the `install` section and remove the macOS only resource while installing into virtualenv.